plugins/modules/docker_swarm_service.py

```diff
@@ -2551,7 +2551,9 @@ def _detect_update_config_failure_action_rollback(client):
     rollback_config_failure_action = (
         (client.module.params['update_config'] or {}).get('failure_action')
     )
-    update_failure_action = client.module.params['update_failure_action']
+    update_failure_action = (
+        client.module.params.get('update_failure_action')
+    )
     failure_action = rollback_config_failure_action or update_failure_action
     return failure_action == 'rollback'
```

Comment:
```suggestion
    return rollback_config_failure_action == 'rollback'
```

Reply:
Added changelog fragment and squashed into one commit.

Above looks like fine to me. Reason I did it the way I did is that, although the option has been removed, there is still a bunch of references to it in the code and in the tests too. Was taking cautious approach, in case there is somehow some legit way for `module.params.update_failure_action` to be set internally.

---
 
```
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:69:  register: update_failure_action_1
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:79:  register: update_failure_action_2
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:89:  register: update_failure_action_3
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:99:  register: update_failure_action_4
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:108:    update_failure_action: "rollback"
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:109:  register: update_failure_action_5
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:120:      - update_failure_action_1 is changed
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:121:      - update_failure_action_2 is not changed
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:122:      - update_failure_action_3 is changed
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:126:      - update_failure_action_4 is changed
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:127:      - update_failure_action_5 is not changed
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:132:      - update_failure_action_4 is failed
tests/integration/targets/docker_swarm_service/tasks/tests/update_config.yml:133:      - "'Minimum version required' in update_failure_action_4.msg"
tests/integration/targets/docker_swarm_service/vars/main.yml:48:  update_failure_action: null
plugins/modules/docker_swarm_service.py:741:    "update_failure_action": null,
plugins/modules/docker_swarm_service.py:1238:        self.update_failure_action = None
plugins/modules/docker_swarm_service.py:1293:            'update_failure_action': self.update_failure_action,
plugins/modules/docker_swarm_service.py:1390:            'update_failure_action': failure_action,
plugins/modules/docker_swarm_service.py:1744:        if self.update_failure_action is not None and self.update_failure_action != os.update_failure_action:
plugins/modules/docker_swarm_service.py:1745:            differences.add('update_failure_action', parameter=self.update_failure_action, active=os.update_failure_action)
plugins/modules/docker_swarm_service.py:1962:        if self.update_failure_action is not None:
plugins/modules/docker_swarm_service.py:1963:            update_config_args['failure_action'] = self.update_failure_action
plugins/modules/docker_swarm_service.py:2151:            ds.update_failure_action = update_config_data.get('FailureAction')
plugins/modules/docker_swarm_service.py:2554:    update_failure_action = (
plugins/modules/docker_swarm_service.py:2555:        client.module.params.get('update_failure_action')
plugins/modules/docker_swarm_service.py:2557:    failure_action = rollback_config_failure_action or update_failure_action
```

Reply:
Actually, there's only one reference to that argument left, and that's in the integration tests and by accident - I'm removing that in #104. So the change I proposed in the above suggestion is safe.

Reply:
#104 -> oh great, thanks. Will just merge then.